### PR TITLE
CEENG-289 Common Settings Component

### PR DESCRIPTION
**Pull request by nikitasheremet**
Created a common settings component between Dashboard and Terminal, by default has 2 cards Account and Info, and can accept 2 more cards.

Desktop:
image
Tablet:
image

### CODE REVIEW

File: src/Inputs/Timepicker/index.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface TimepickerProps extends LabelLayoutProps {
     onChange?: Function;
     disabled?: boolean;
 }
-
+// @ts-ignore not sure why this is being highlighted as an issue
```
Comment (ralph-dev) [inline src/Inputs/Timepicker/index.tsx:16]: Was line 25 being highlighted? I wonder if that is the issue

Comment (nikitasheremet) [reply src/Inputs/Timepicker/index.tsx:16]: Let me look at the error

File: src/Containers/Settings.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    width: 100%;
+`;
+
+interface SettingsProps
+    extends MainInterface,
+        ResponsiveInterface,
+        InfoProps,
+        AccountProps {
+    loading: boolean;
+    cardOne?: React.ReactElement;
```
Comment (fpunny) [inline src/Containers/Settings.tsx:32]: Give it better names like info, account, etc.

File: src/Containers/Settings.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+interface AccountProps extends MainInterface, ResponsiveInterface {
+    employee: { username: string; role: string; _id: string };
+    logout: Function;
+}
+interface DisplayItemProps extends MainInterface, ResponsiveInterface {
+    label: string;
+    value: string;
+}
+
+const Info: React.FC<InfoProps> = ({ faqName, faqLink, version }) => {
```
Comment (fpunny) [inline src/Containers/Settings.tsx:50]: Something i failed to do, which we need for every component to fix most of our TS issues is to inlude the props from the actual html element which the component is based off
https://stackoverflow.com/questions/58924179/typscript-react-extending-default-functional-component-props-gives-error

File: src/Containers/Settings.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    const state = useState(false);
+    return (
+        <SettingsCard heading="Additional Information" icon={InfoCircle}>
+            <Buttons>
+                <Button
+                    margin="5px"
+                    icon={Book}
+                    onClick={() =>
+                        window.open(
```
Comment (fpunny) [inline src/Containers/Settings.tsx:59]: If we have time, we should make an LinkButton instead

File: src/Containers/Settings.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                <Paragraph margin="15px 0" bold>
+                    {'If you require immediate assistance, please call/text: '}
+                    <a href="tel:13063411035">+1 (306) 341-1035</a>
+                </Paragraph>
+                <Button onClick={() => state[1](false)}>Close</Button>
+            </Modal>
+        </SettingsCard>
+    );
+};
+const Account: React.FC<AccountProps> = ({ employee, logout }) => {
```
Comment (fpunny) [inline src/Containers/Settings.tsx:104]: This is unholy. pls break this into multiple files

File: src/Containers/Settings.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        <>
+            <Loading loading={loading} inlineStyle={loadingStyle} {...props}>
+                <div>
+                    <Heading type="h1" bold lineHeight="1.2" margin="0 0 10px">
+                        Terminal Settings
+                    </Heading>
+                    {!loading && (
+                        <Cards>
+                            <Account employee={employee} logout={logout} />
+                            {cardTwo && cardTwo}
```
Comment (fpunny) [inline src/Containers/Settings.tsx:167]: I am confusion

File: src/Containers/Settings.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    cardThree,
+    faqName = 'FAQ',
+    faqLink,
+    version,
+    employee = { username: '', role: '', _id: '' },
+    logout,
+    ...props
+}): React.ReactElement => {
+    return (
+        <>
```
Comment (fpunny) [inline src/Containers/Settings.tsx:158]: Fragment is useless

File: src/Containers/Copyright.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import { SmallText } from '../Text/SmallText';
+
+interface CopyrightProps extends MainInterface, ResponsiveInterface {
+    margin: string;
+    version: string | number;
+}
+
+export const Copyright: React.FC<CopyrightProps> = ({
+    margin,
+    version,
```
Comment (fpunny) [inline src/Containers/Copyright.tsx:12]: Better yet, make version a theme field

Comment (fpunny) [inline src/Containers/Copyright.tsx:12]: default this to npm package version

Comment (nikitasheremet) [reply src/Containers/Copyright.tsx:12]: Wouldn't this just default to the react-ui package version then?

File: src/Containers/Settings.tsx (hunk 3 above)
Comment (nikitasheremet) [reply src/Containers/Settings.tsx:50]: So we need to include the html props for every type of element that is present in the component, or the props for the outermost element (in the code above, the outermost element is SettingsCard which is a div)?

File: src/Inputs/Timepicker/index.tsx [hunk 9]
```diff
@@ -13,7 +13,7 @@ export interface TimepickerProps extends LabelLayoutProps {
     onChange?: Function;
     disabled?: boolean;
 }
-// @ts-ignore not sure why this is being highlighted as an issue
+// @ts-ignore TimePicker throwing error because of withTheme
```
Comment (ralph-dev) [inline src/Inputs/Timepicker/index.tsx:16]: https://github.com/mui-org/material-ui/issues/8070 does this solve it?

File: src/Containers/Settings.tsx (hunk 3 above)
Comment (fpunny) [reply src/Containers/Settings.tsx]: outmost

File: src/Containers/Copyright.tsx (hunk 8 above)
Comment (fpunny) [reply src/Containers/Copyright.tsx:12]: hmm ur right, nvm then :/

File: src/Inputs/Timepicker/index.tsx (hunk 9 above)
Comment (nikitasheremet) [reply src/Inputs/Timepicker/index.tsx:16]: @fpunny Could you explain what withTheme is being used for in Timepicker?

File: src/Containers/Loading.tsx [hunk 10]
```diff
@@ -9,7 +9,10 @@ import {
 } from '@Utils/BaseStyles';
 import { useTransition } from '@Utils/Hooks';
 
-export interface LoadingProps extends ResponsiveInterface, MainInterface {
+export interface LoadingProps
+    extends React.HTMLAttributes<HTMLDivElement>,
```
Comment (fpunny) [inline src/Containers/Loading.tsx:13]: LGTM owo

**Pull request merged**